core/src/com/unciv/ui/worldscreen/mainmenu/OptionsPopup.kt

```diff
@@ -337,13 +338,32 @@ class OptionsPopup(val previousScreen: BaseScreen) : Popup(previousScreen) {
             BaseScreen.enableSceneDebug = it
         }).row()
         val unlockTechsButton = "Unlock all techs".toTextButton()
-        unlockTechsButton.onChange {
-            for (tech in game.gameInfo.ruleSet.technologies.keys) {
-                if (tech !in game.gameInfo.getCurrentPlayerCivilization().tech.techsResearched)
-                    game.gameInfo.getCurrentPlayerCivilization().tech.addTechnology(tech)
+        unlockTechsButton.onClick {
+            if (game.isGameInfoInitialized()) {
+                for (tech in game.gameInfo.ruleSet.technologies.keys) {
+                    if (tech !in game.gameInfo.getCurrentPlayerCivilization().tech.techsResearched) {
+                        game.gameInfo.getCurrentPlayerCivilization().tech.addTechnology(tech)
+                        game.gameInfo.getCurrentPlayerCivilization().popupAlerts.removeLastOrNull()
+                    }
+                }
+                game.gameInfo.getCurrentPlayerCivilization().updateSightAndResources()
+                game.worldScreen.shouldUpdate = true
+            }
+        }
+        add(unlockTechsButton).row()
+        val giveResourcesButton = "Give all strategic resources".toTextButton()
```

Comment:
"Improve", no? "Give all" makes it seem like you're giving resources not on the map

Reply:
Well, the resource deposits aren't on the map until after the player presses the button. IMO "Improve" sounds like it will only put improvements on what you already have.